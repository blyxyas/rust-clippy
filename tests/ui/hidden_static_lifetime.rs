#![allow(unused)]
#![warn(clippy::hidden_static_lifetime)]
#![allow(clippy::needless_lifetimes)]

struct S<'s> {
    s: &'s u32,
}

// Should warn

fn a<'a>() -> &'a str {
    ""
}
fn h<'h>() -> S<'h> {
    S { s: &1 }
}

// Should not warn
fn b<'b>(_: &'b str) -> &'b str {
    ""
}
fn d<'d>(_: &'d str) {}
fn e<'e>(_: &'e str) -> &'e str {
    ""
}
fn f<'f, F>(_: F) -> F
where
    F: 'f,
{
    todo!()
}
fn g<'g>(_: S<'g>) -> S<'g> {
    S { s: &1 }
}

fn i() -> S<'static> {
    S { s: &1 }
}

fn main() {}
