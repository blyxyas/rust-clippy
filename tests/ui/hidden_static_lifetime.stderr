error: This lifetime can be changed to `'static`
  --> $DIR/hidden_static_lifetime.rs:11:6
   |
LL | fn a<'a>() -> &'a str {
   |      ^^ help: try: `changing this lifetime to 'static`
   |
   = note: `-D clippy::hidden-static-lifetime` implied by `-D warnings`

error: This lifetime can be changed to `'static`
  --> $DIR/hidden_static_lifetime.rs:14:6
   |
LL | fn h<'h>() -> S<'h> {
   |      ^^ help: try: `changing this lifetime to 'static`

error: aborting due to 2 previous errors

